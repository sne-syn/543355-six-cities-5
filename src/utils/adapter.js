export const adaptOffer = (data) => {
  return {
    bedrooms: data[`bedrooms`],
    city: {
      location: adaptLocation(data[`city`][`location`]),
      name: data[`city`][`name`]
    },
    description: data[`description`],
    goods: data[`goods`],
    host: adaptHost(data[`host`]),
    id: data[`id`],
    images: data[`images`],
    isFavorite: Boolean(data[`is_favorite`]),
    isPremium: Boolean(data[`is_favorite`]),
    location: adaptLocation(data[`location`]),
    maxGuests: data[`max_adults`],
    previewImage: data[`preview_image`],
    price: data[`price`],
    rating: data[`rating`],
    title: data[`title`],
    type: data[`type`],
  };
};

const adaptLocation = (location) => {
  return {
    latitude: location[`latitude`],
    longitude: location[`longitude`],
    zoom: location[`zoom`]
  };
};

const adaptHost = (host) => {
  return {
    avatar: host[`avatar_url`],
    id: host[`id`],
    isPro: Boolean(host[`is_pro`]),
    name: host[`name`]
  };
};

export const adaptReview = (review) => {
  return {
    author: review[`user`][`name`],
    avatar: review[`user`][`avatar_url`],
    date: review[`date`],
    id: review[`id`],
    isPro: Boolean(review[`is_pro`]),
    rating: review[`rating`],
    comment: review[`comment`],
    userId: review[`user`][`id`]
  };
};

export const adaptOffers = (offers) => {
  return offers.map((offer) => adaptOffer(offer));
};

export const adaptReviews = (reviews) => {
  return reviews.map((review) => adaptReview(review));
};
