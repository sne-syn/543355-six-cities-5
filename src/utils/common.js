const AccomodationTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

// add leading zero to num < 10
const addLeadingZero = (value) => {
  return (value < 10) ? `0${value}` : value;
};

const generateID = function () {
  return `_${Math.random().toString(36).substr(2, 9)}`;
};

// generate randome integer
const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

// generate random non-integer
const getRandomNumber = (min, max) => {
  return min + (Math.random() * (max - min));
};

// get 1 random item from arrayOfItems
const getRandomArrayItem = (array) => {
  const randomIndex = getRandomIntegerNumber(0, array.length);
  return array[randomIndex];
};

// get several random items
const getSeveralRandomItems = (array, count) => {
  let set = new Set();
  for (let i = 0; i < count; i++) {
    set.add(getRandomArrayItem(array));
  }
  return [...set];
};

// capitalize first char only
const capitalizeChar = (str) => {
  if (typeof str !== `string`) {
    return ``;
  }
  return str.charAt(0).toUpperCase() + str.slice(1);
};

export {
  AccomodationTypes,
  addLeadingZero,
  capitalizeChar,
  generateID,
  getRandomIntegerNumber,
  getRandomArrayItem,
  getRandomNumber,
  getSeveralRandomItems,
};
