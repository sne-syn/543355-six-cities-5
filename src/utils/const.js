const NEAR_PLACES_COUNT = 3;
const ComponentType = {
  CITIES: `cities`,
  NEAR: `near-places`,
  FAVORITE: `favorites`
};

const SortType = new Map();
SortType.set(`DEFAULT`, `Popular`);
SortType.set(`PRICE_LOW_HIGH`, `Price: low to high`);
SortType.set(`PRICE_HIGH_LOW`, `Price: high to low`);
SortType.set(`TOP_RATED`, `Top rated first`);

const AccomodationTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

const MAX_RATING = 5;
const CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const MONTH_NAMES = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

export {NEAR_PLACES_COUNT, MAX_RATING, CITIES, MONTH_NAMES, AccomodationTypes, ComponentType, SortType};
