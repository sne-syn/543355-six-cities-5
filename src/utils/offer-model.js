export const adaptOffer = (data) => {
  return {
    id: data[`id`],
    city: {
      location: adaptLocation(data[`city`][`location`]),
      name: data[`city`][`name`]
    },
    title: data[`title`],
    images: data[`images`],
    previewImage: data[`preview_image`],
    price: data[`price`],
    type: data[`type`],
    rating: data[`rating`],
    goods: data[`goods`],
    bedrooms: data[`bedrooms`],
    maxGuests: data[`max_adults`],
    description: data[`description`],
    isPremium: Boolean(data[`is_favorite`]),
    isFavorite: Boolean(data[`is_favorite`]),
    host: adaptHost(data[`host`]),
    location: adaptLocation(data[`location`])
  };
};

const adaptLocation = (location) => {
  return {
    latitude: location[`latitude`],
    longitude: location[`longitude`],
    zoom: location[`zoom`]
  };
};

const adaptHost = (host) => {
  return {
    id: host[`id`],
    avatar: host[`avatar_url`],
    name: host[`name`],
    isPro: Boolean(host[`is_pro`]),
  };
};

export const adaptReview = (review) => {
  return {
    id: review[`id`],
    avatar: review[`user`][`avatar_url`],
    text: review[`comment`],
    author: review[`user`][`name`],
    date: review[`date`],
    rating: review[`rating`],
    isPro: Boolean(review[`is_pro`]),
    userId: review[`user`][`id`]
  };
};

export const adaptOffers = (offers) => {
  return offers.map((offer) => adaptOffer(offer));
};

export const adaptReviews = (reviews) => {
  return reviews.map((review) => adaptReview(review));
};
