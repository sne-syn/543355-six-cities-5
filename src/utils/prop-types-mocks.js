import PropTypes from 'prop-types';

export const hostPropTypesMock = {
  id: PropTypes.number.isRequired,
  avatar: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  isPro: PropTypes.bool.isRequired
};

export const offerPropTypesMock = PropTypes.shape({
  id: PropTypes.number.isRequired,
  city: PropTypes.shape({
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired,
    }).isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  title: PropTypes.string.isRequired,
  previewImage: PropTypes.string.isRequired,
  images: PropTypes.array.isRequired,
  price: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  goods: PropTypes.array.isRequired,
  bedrooms: PropTypes.number.isRequired,
  maxGuests: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  host: PropTypes.shape(hostPropTypesMock).isRequired,
  isPremium: PropTypes.bool.isRequired,
  isFavorite: PropTypes.bool.isRequired,
}).isRequired;

export const offerItemPropTypesMock = PropTypes.shape({
  id: PropTypes.number,
  city: PropTypes.shape({
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
    name: PropTypes.string,
  }),
  title: PropTypes.string,
  images: PropTypes.array,
  price: PropTypes.number,
  type: PropTypes.string,
  rating: PropTypes.number,
  goods: PropTypes.array,
  bedrooms: PropTypes.number,
  maxGuests: PropTypes.number,
  description: PropTypes.string,
  host: PropTypes.shape(hostPropTypesMock),
  isPremium: PropTypes.bool,
  isFavorite: PropTypes.bool,
}).isRequired;
