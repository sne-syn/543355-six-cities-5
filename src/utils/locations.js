import {getRandomIntegerNumber} from '../utils/common';

export const LocationsList = {
  Amsterdam: [
    [52.3909553943508, 4.85309666406198],
    [52.369553943508, 4.85309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3809553943508, 4.939309666406198],
    [52.4006614685059, 4.88902187347412],
    [52.3763414, 4.8999733],
    [52.333889, 4.888803]
  ], Dusseldorf: [
    [51.237402, 6.797314],
    [51.239402000000005, 6.756314000000001],
    [51.248402000000006, 6.763314],
    [51.205402, 6.7613140000000005]
  ], Cologne: [
    [50.947361, 6.9799739999999995],
    [50.949361, 6.976974],
    [50.932361, 6.960974],
    [50.945361, 6.935974],
    [50.913361, 6.9509739999999995]
  ], Brussels: [
    [50.822556999999996, 4.347697],
    [50.844556999999995, 4.346697],
    [50.869557, 4.332697],
    [50.852557, 4.3376969999999995],
    [50.842557, 4.363696999999999]
  ], Hamburg: [
    [53.563341, 10.019654000000001],
    [53.574341000000004, 10.022654000000001]
  ], Paris: [
    [48.83961, 2.342499],
    [48.865610000000004, 2.350499],
    [48.84461, 2.374499],
    [48.861610000000006, 2.340499]
  ]
};

export const CoordinatesMap = {
  AMSTERDAM: {
    latitude: 52.38333,
    longitude: 4.9,
    zoom: getRandomIntegerNumber(10, 14)
  },
  DUSSELDORF: {
    latitude: 51.225402,
    longitude: 6.776314,
    zoom: getRandomIntegerNumber(10, 14)
  },
  COLOGNE: {
    latitude: 50.938361,
    longitude: 6.959974,
    zoom: getRandomIntegerNumber(11, 14)
  },
  HAMBURG: {
    latitude: 53.550341,
    longitude: 10.000654,
    zoom: getRandomIntegerNumber(10, 14)
  },
  PARIS: {
    latitude: 48.85661,
    longitude: 2.351499,
    zoom: getRandomIntegerNumber(11, 14)
  },
  BRUSSELS: {
    latitude: 50.846557,
    longitude: 4.351697,
    zoom: getRandomIntegerNumber(10, 14)
  },
};
