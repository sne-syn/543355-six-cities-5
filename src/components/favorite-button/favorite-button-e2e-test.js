import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {offer} from '../../test-data/offer-test-data';
import {ComponentType} from '../../utils/const';
import {FavoriteButton} from './favorite-button';

configure({adapter: new Adapter()});

const mockEvent = {
  preventDefault() {}
};

it(`Should callback be click actions`, () => {
  const handleFavoriteButtonClickAction = jest.fn();
  const handleRedirectToLoginAction = jest.fn();
  const handleToggleComponent = jest.fn();
  const handleUpdateOfferItemInStoreAction = jest.fn();
  const handleUpdateOffersInStoreAction = jest.fn();
  const handleUpdateNearPlacesInStoreAction = jest.fn();

  const wrapper = shallow(
      <FavoriteButton
        authorizationStatus={`NO_AUTH`}
        componentName={`place-card`}
        defaultOnValue={true}
        offerId={1}
        nearPlaces={[]}
        offerItem={{}}
        offers={[]}
        on={true}
        onFavoriteButtonClickAction={handleFavoriteButtonClickAction}
        redirectToLoginAction={handleRedirectToLoginAction}
        toggleComponent={handleToggleComponent}
        updateOfferItemInStoreAction={handleUpdateOfferItemInStoreAction}
        updateOffersInStoreAction={handleUpdateOffersInStoreAction}
        updateNearPlacesInStoreAction={handleUpdateNearPlacesInStoreAction}
      />
  );

  const favoriteButton = wrapper.find(`.button`);

  // favoriteButton.simulate(`click`, mockEvent);
  // expect(handleFavoriteButtonClickAction).toHaveBeenCalledTimes(1);

  // favoriteButton.simulate(`click`, mockEvent);
  // expect(handleRedirectToLoginAction).toHaveBeenCalledTimes(1);
  // favoriteButton.simulate(`click`, mockEvent);
  // expect(handleToggleComponent).toHaveBeenCalledTimes(1);
  // favoriteButton.simulate(`click`, mockEvent);
  // expect(handleUpdateOfferItemInStoreAction).toHaveBeenCalledTimes(1);
  // favoriteButton.simulate(`click`, mockEvent);
  // expect(handleUpdateOffersInStoreAction).toHaveBeenCalledTimes(1);
  // favoriteButton.simulate(`click`, mockEvent);
  // expect(handleUpdateNearPlacesInStoreAction).toHaveBeenCalledTimes(1);
});

import React from "react";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {OfferBookmarkBtn} from "./offer-bookmark-btn";
import {mockOffer} from "../../utils";
import {AuthStatus} from "../../constants";

configure({adapter: new Adapter()});

describe(`OfferBookmarkBtn e2e tests`, () => {
  test(`Click on OfferBookmarkBtn AuthStatus.NO_AUTH`, () => {
    const bookmarkBtnClickAction = jest.fn();

    const wrapper = shallow(
        <OfferBookmarkBtn
          offer={mockOffer}
          bookmarkType={`MAIN`}
          bookmarkBtnClickAction={bookmarkBtnClickAction}
          authStatus={AuthStatus.NO_AUTH}
        />
    );

    wrapper.find(`.button`).simulate(`click`, {preventDefault() {}});
    expect(bookmarkBtnClickAction).toHaveBeenCalledTimes(0);
  });

  test(`Click on OfferBookmarkBtn AuthStatus AUTH`, () => {
    const bookmarkBtnClickAction = jest.fn();

    const wrapper = shallow(
        <OfferBookmarkBtn
          offer={mockOffer}
          bookmarkType={`MAIN`}
          bookmarkBtnClickAction={bookmarkBtnClickAction}
          authStatus={AuthStatus.AUTH}
        />
    );

    wrapper.find(`.button`).simulate(`click`, {preventDefault() {}});
    expect(bookmarkBtnClickAction).toHaveBeenCalledTimes(1);
  });
});


describe(`FavoriteButton behavior`, () => {
  it(`when authorization status is AUTH`, () => {

    const redirectToLoginPage = jest.fn();
    const changeOfferStatus = jest.fn();
    const authorizationStatus = AuthorizationStatus.AUTH;

    const wrapper = shallow(
        <FavoriteButton
          isFavorite={ true }
          className={ FavoriteButtonProperty.Property.CLASS_NAME }
          buttonWidth={ FavoriteButtonProperty.Property.WIDTH }
          buttonHeight={ FavoriteButtonProperty.Property.WIDTH }
          authorizationStatus={ authorizationStatus }
          id={ 1 }
          redirectToLoginPage={ redirectToLoginPage }
          changeOfferStatus={ changeOfferStatus }
        />
    );

    const button = wrapper.find(`button`);

    button.simulate(`click`);

    expect(changeOfferStatus).toHaveBeenCalledTimes(1);
    expect(redirectToLoginPage).toHaveBeenCalledTimes(0);

  });

  it(`when authorization status is NO_AUTH`, () => {

    const redirectToLoginPage = jest.fn();
    const changeOfferStatus = jest.fn();
    const authorizationStatus = AuthorizationStatus.NO_AUTH;

    const wrapper = shallow(
        <FavoriteButton
          isFavorite={ true }
          className={ FavoriteButtonProperty.Property.CLASS_NAME }
          buttonWidth={ FavoriteButtonProperty.Property.WIDTH }
          buttonHeight={ FavoriteButtonProperty.Property.WIDTH }
          authorizationStatus={ authorizationStatus }
          id={ 1 }
          redirectToLoginPage={ redirectToLoginPage }
          changeOfferStatus={ changeOfferStatus }
        />
    );

    const button = wrapper.find(`button`);

    button.simulate(`click`);

    expect(changeOfferStatus).toHaveBeenCalledTimes(0);
    expect(redirectToLoginPage).toHaveBeenCalledTimes(1);

  });
});
