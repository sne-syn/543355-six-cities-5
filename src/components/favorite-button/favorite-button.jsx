import PropTypes from 'prop-types';
import React, {PureComponent} from 'react';
import withToggler from '../../HOCs/with-toggler';
import {AppRoute, AuthorizationStatus} from '../../utils/const';
import {addToFavorites} from '../../store/api-actions';
import {redirectToRoute, updateOffersInStore} from '../../store/action';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../store/user-data/user-data-selectors';
import {getOffers} from '../../store/offers-data/offers-data-selectors';

const getIconSize = (componentName) => {
  return (componentName === `property`) ? {
    width: 31,
    height: 33,
  } :
    {
      width: 18,
      height: 19,
    };
};

class FavoriteButton extends PureComponent {
  constructor(props) {
    super(props);
    this._handleAddToFavorite = this._handleAddToFavorite.bind(this);
  }

  _handleAddToFavorite(id) {
    const {updateOffersInStoreAction, onFavoriteButtonClickAction, offers} = this.props;
    const isFavorite = !this.props.on;
    if (offers !== 0) {
      const offerShouldUpdate = offers.find((offer) => offer.id === id);
      const newOffer = Object.assign({}, offerShouldUpdate, {isFavorite: !this.props.on});
      updateOffersInStoreAction(newOffer, id);
    }
    onFavoriteButtonClickAction(id, Number(isFavorite));
  }

  render() {
    const {authorizationStatus, componentName, offerId, on, redirectToLoginAction, toggleComponent} = this.props;
    const isLogged = authorizationStatus === AuthorizationStatus.AUTH;
    let buttonClass = `${componentName}__bookmark-button button `;
    buttonClass += (on) ? `${componentName}__bookmark-button--active` : ``;
    const iconSize = getIconSize(componentName);
    const favoriteButtonPlaceholder = `${on ? (`In`) : (`To`)} bookmarks`;
    const onClickActions = isLogged ? () => {
      toggleComponent(); this._handleAddToFavorite(offerId);
    } : redirectToLoginAction;

    return (
      <button className={buttonClass} onClick={onClickActions} type="button">
        <svg className={`place-card__bookmark-icon`} width={`${iconSize.width}`} height={`${iconSize.height}`}>
          <use xlinkHref="#icon-bookmark"></use>
        </svg>
        <span className="visually-hidden">{favoriteButtonPlaceholder}</span>
      </button>
    );
  }
}

FavoriteButton.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  componentName: PropTypes.string.isRequired,
  offerId: PropTypes.number.isRequired,
  offers: PropTypes.array.isRequired,
  on: PropTypes.bool.isRequired,
  onFavoriteButtonClickAction: PropTypes.func.isRequired,
  redirectToLoginAction: PropTypes.func.isRequired,
  toggleComponent: PropTypes.func.isRequired,
  updateOffersInStoreAction: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  offers: getOffers(state),
});

const mapDispatchToProps = (dispatch) => ({
  onFavoriteButtonClickAction(id, status) {
    dispatch(addToFavorites(id, status));
  },
  updateOffersInStoreAction(offer) {
    dispatch(updateOffersInStore(offer));
  },
  redirectToLoginAction() {
    dispatch(redirectToRoute(AppRoute.LOGIN));
  }
});

export {FavoriteButton};
export default connect(mapStateToProps, mapDispatchToProps)(withToggler(FavoriteButton));
