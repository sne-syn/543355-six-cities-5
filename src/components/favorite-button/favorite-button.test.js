import React from 'react';
import renderer from 'react-test-renderer';
import {AuthorizationStatus} from '../../utils/const';
import {FavoriteButton} from '../favorite-button/favorite-button';

const noop = () => {};

describe(`<FavoriteButton /> test`, () => {
  it(`renders FavoriteButton active`, () => {
    const tree = renderer
    .create(
        <FavoriteButton
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          componentName={`place-card`}
          defaultOnValue={true}
          offerId={1}
          offers={[]}
          on={true}
          onFavoriteButtonClickAction={noop}
          redirectToLoginAction={noop}
          toggleComponent={noop}
          updateOffersInStoreAction={noop} />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders FavoriteButton no-active`, () => {
    const tree = renderer
    .create(

        <FavoriteButton
          authorizationStatus={AuthorizationStatus.AUTH}
          componentName={`place-card`}
          defaultOnValue={false}
          offerId={1}
          offers={[]}
          on={false}
          onFavoriteButtonClickAction={noop}
          redirectToLoginAction={noop}
          toggleComponent={noop}
          updateOffersInStoreAction={noop} />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders FavoriteButton for property-place`, () => {
    const tree = renderer
    .create(

        <FavoriteButton
          authorizationStatus={AuthorizationStatus.AUTH}
          componentName={`property`}
          defaultOnValue={false}
          offerId={1}
          offers={[]}
          on={false}
          onFavoriteButtonClickAction={noop}
          redirectToLoginAction={noop}
          toggleComponent={noop}
          updateOffersInStoreAction={noop} />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

