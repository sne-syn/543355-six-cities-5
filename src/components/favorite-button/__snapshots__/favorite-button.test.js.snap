// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteButton /> test renders FavoriteButton active 1`] = `
<button
  className="place-card__bookmark-button button place-card__bookmark-button--active"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`<FavoriteButton /> test renders FavoriteButton for property-place 1`] = `
<button
  className="property__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`<FavoriteButton /> test renders FavoriteButton no-active 1`] = `
<button
  className="place-card__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
